src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -0,0 +1,221 @@
+   *
+   */
+  constructor(props) {
+    assert(props.hexagonVertices, 'hexagonVertices must be supplied');
+    super(Object.assign({}, defaultProps, props));
+  }
+
+  /**
+   * DeckGL calls initializeState when GL context is available
+   * Essentially a deferred constructor
+   */
+  initializeState() {
+    const {gl} = this.context;
+    this.setState({model: this.getModel(gl)});
+
+    const {attributeManager} = this.state;
+    attributeManager.addInstanced({
+      instancePositions: {size: 3, update: this.calculateInstancePositions},
+      instanceColors: {size: 4, update: this.calculateInstanceColors}
+    });
+
+    this.updateRadiusAngle();
+  }
+
+  updateState({props, oldProps, changeFlags: {dataChanged, viewportChanged}} = {}) {
+    const {model, attributeManager} = this.state;
+
+    if (dataChanged) {
+      attributeManager.invalidateAll();
+    }
+
+    // Update the positions in the model if they've changes
+    const verticesChanged =
+      !positionsAreEqual(oldProps.hexagonVertices, props.hexagonVertices);
+
+    if (model && (verticesChanged || viewportChanged)) {
+      this.updateRadiusAngle();
+    }
+
+    this.updateUniforms();
+  }
+
+  updateRadiusAngle() {
+    const {hexagonVertices: vertices, dotRadius} = this.props;
+
+    const vertex0 = vertices[0];
+    const vertex3 = vertices[3];
+
+    // transform to space coordinates
+    const spaceCoord0 = this.projectFlat(vertex0);
+    const spaceCoord3 = this.projectFlat(vertex3);
+
+    // distance between two close centroids
+    const dx = spaceCoord0[0] - spaceCoord3[0];
+    const dy = spaceCoord0[1] - spaceCoord3[1];
+    const dxy = Math.sqrt(dx * dx + dy * dy);
+
+    this.setUniforms({
+      // Calculate angle that the perpendicular hexagon vertex axis is tilted
+      angle: Math.acos(dx / dxy) * -Math.sign(dy) + Math.PI / 2,
+      // Allow user to fine tune radius
+      radius: dxy / 2 * Math.min(1, dotRadius)
+    });
+  }
+
+  getCylinderGeometry(radius) {
+    return new CylinderGeometry({
+      radius,
+      topRadius: radius,
+      bottomRadius: radius,
+      topCap: true,
+      bottomCap: true,
+      height: 1,
+      nradial: 6,
+      nvertical: 1
+    });
+  }
+
+  updateUniforms() {
+    const {opacity, enable3d, elevation, invisibleColor} = this.props;
+    this.setUniforms({
+      opacity,
+      enable3d: enable3d ? 1 : 0,
+      elevation,
+      invisibleColor: [invisibleColor.r, invisibleColor.g, invisibleColor.b]
+    });
+  }
+
+  getShaders() {
+    const vertex = readFileSync(join(__dirname, './hexagon-layer-vertex.glsl'), 'utf8');
+    const lighting = readFileSync(join(__dirname, './lighting.glsl'), 'utf8');
+    const picking = readFileSync(join(__dirname, './picking.glsl'), 'utf8');
+    const vs = picking.concat(lighting).concat(vertex);
+
+    return {
+      vs,
+      fs: readFileSync(join(__dirname, './hexagon-layer-fragment.glsl'), 'utf8')
+    };
```

Comment:
About to merge a PR where you can just add 

    modules: ['lighting']

to the getShaders object.